### PR TITLE
Update release_process.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Create draft.

### CODE REVIEW

**Comment by AnotherButler** [comment]
@adbridge Thanks for sharing the additional content. I've incorporated some of it, so I think this is ready for your review now.

File: Docs/release_process.md [hunk 1]
```diff
@@ -1,9 +1,13 @@
 ## How We Release Arm Mbed OS
 
-The three types of Arm Mbed OS releases are major releases, feature releases and patch releases. Major releases happen infrequently and indicate a change in the structure of the OS. In major releases, the first number after "Mbed OS" changes. For example, moving to Arm Mbed OS 5.0 was a major release. Feature releases occur about once a quarter. As their name suggests, feature releases introduce new features to the code base. The second digit in the release number indicates the feature release. For example, Mbed OS 5.1.0 indicates major version 5, feature version 1. Patch releases occur about once every two weeks. They include bug fixes and additional target support. The last digit in the release number indicates the patch release. In Mbed OS 5.2.3, `3` shows the patch release is the third release in Mbed 5.2.
+The three types of Arm Mbed OS releases are major releases, feature releases and patch releases. Major releases happen infrequently and indicate a change in the structure of the OS. In major releases, the first number after "Mbed OS" changes. For example, moving to Arm Mbed OS 5.0 was a major release.
```
Comment (adbridge) [inline Docs/release_process.md]: The 3 types are Major, Minor and Patch. Minor release could be thought of as feature releases, but we should get the terminology correct. So I would rephrase the paragraph to something like:

The three type of Arm Mbed OS releases are major, minor and patch. Major releases happen infrequently and indicate a change in the structure of the OS. In major releases, the first number after "Mbed OS" changes. For example, moving to Arm Mbed OS 5.0 was a major release.

File: Docs/release_process.md [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+The three types of Arm Mbed OS releases are major releases, feature releases and patch releases. Major releases happen infrequently and indicate a change in the structure of the OS. In major releases, the first number after "Mbed OS" changes. For example, moving to Arm Mbed OS 5.0 was a major release. 
+
+Feature releases occur about once a quarter. As their name suggests, feature releases introduce new features to the code base. Unlike major releases, feature releases are backward compatible. The second digit in the release number indicates the feature release. For example, Mbed OS 5.1.0 indicates major version 5, feature version 1.
```
Comment (adbridge) [inline Docs/release_process.md]: Minor releases occur once a quarter and introduce new features to the code base. Unlike major releases, minor releases are backward compatible. The second digit in the release number indicates the minor release. For example, Mbed OS 5.1.0 indicates major version 5, minor version 1.

File: Docs/release_process.md [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
+The three types of Arm Mbed OS releases are major releases, feature releases and patch releases. Major releases happen infrequently and indicate a change in the structure of the OS. In major releases, the first number after "Mbed OS" changes. For example, moving to Arm Mbed OS 5.0 was a major release. 
+
+Feature releases occur about once a quarter. As their name suggests, feature releases introduce new features to the code base. Unlike major releases, feature releases are backward compatible. The second digit in the release number indicates the feature release. For example, Mbed OS 5.1.0 indicates major version 5, feature version 1. 
+
+Patch releases occur about once every two weeks. They include bug fixes and additional target support. They are also backward compatible. The last digit in the release number indicates the patch release. In Mbed OS 5.2.3, `3` shows the patch release is the third release in Mbed 5.2.
```
Comment (adbridge) [inline Docs/release_process.md]: Patch releases occur every two weeks. They include bug fixes, new target support and improvements to existing functionality.

File: Docs/release_process.md [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
 
 ### The release process
 
-Every release goes through a rigorous review and testing process. Two weeks before each release, we implement a code freeze. No new code can go into the release branch after that time. Any new patches must go into a later release. During code freeze, we apply [our testing tools](testing.md) to the release branch. These tests include nightly runs through our continuous integration (CI) processes, as well as out-of-box QA.
+Every release goes through a rigorous review and testing process. We stage changes to the `release-candidate` branch in the `mbed-os` repository. Two weeks before each release, we implement a code freeze. No new code can go into the release branch after that time. Any new patches must go into a later release. During code freeze, we apply [our testing tools](testing.md) to the release branch. These tests include nightly runs through our continuous integration (CI) processes, as well as out-of-box QA. We also put our examples through this testing process.
```
Comment (adbridge) [inline Docs/release_process.md]: Which release are you talking about here? Minor releases have a different process to patch releases ? The 2 week code freeze applies only to minor releases and allows us to run Out Of Box testing. For patch releases code freeze is on the Thu of an even week and the release is on the Wed of an odd week. There is less testing for a patch release, but exporters and nightly are still run.

File: Docs/release_process.md [hunk 5]
```diff
@@ -1,9 +1,13 @@
...
 
-After all tests return no errors, we release the latest updates. A release note accompanies each release. The release notes for major and feature releases are longer and give an overview of the new features. The release notes for the patch releases include only a list of changes and known issues, if applicable.
+After all tests return no errors, we release the latest updates. You can find the most recent release in the `mbed-os` repository with the `latest` tag. A release note accompanies each release. The release notes for major and feature releases are longer and give an overview of the new features. The release notes for the patch releases include only a list of changes and known issues, if applicable. You can find our release notes here and on [our blog](https://developer.mbed.org/blog/).
```
Comment (adbridge) [inline Docs/release_process.md]: Use 'minor release' not 'feature release'.

**Comment by AnotherButler** [comment]
@adbridge Thanks so much for the review. I recently received instructions that we are to start calling all minor releases "feature releases", so that's why I used that terminology. I'll make the other changes shortly. Thanks again.

**Comment by AnotherButler** [comment]
@adbridge I think I've addressed your comments. Please review at your convenience.

**Comment by adbridge** [comment]
LGTM

File: Docs/release_process.md [hunk 6]
```diff
@@ -1,13 +1,19 @@
...
 
 ### The release process
 
 Every release goes through a rigorous review and testing process. We stage changes to the `release-candidate` branch in the `mbed-os` repository. Two weeks before each release, we implement a code freeze. No new code can go into the release branch after that time. Any new patches must go into a later release. During code freeze, we apply [our testing tools](testing.md) to the release branch. These tests include nightly runs through our continuous integration (CI) processes, as well as out-of-box QA. We also put our examples through this testing process.
 
+Every release goes through a rigorous review and testing process. We stage changes to the `release-candidate` branch in the `mbed-os` repository. 
+
+Two weeks before each feature release, we implement a code freeze. No new code can go into the release branch after that time. Any new patches must go into a later release. During this code freeze, we apply [our testing tools](testing.md) to the release branch. These tests include nightly runs through our continuous integration (CI) processes, as well as out-of-box QA. We also put our examples through this testing process.
```
Comment (adbridge) [inline Docs/release_process.md]: We should probably say no new code except for bug fixes found during our OOB testing

Comment (adbridge) [inline Docs/release_process.md]: Also 'Any new patches...' is misleading, I would remove that sentence

Comment (adbridge) [inline Docs/release_process.md]: Maybe a comment like:

Comment (AnotherButler) [reply Docs/release_process.md]: Good points. I like this wording. I've implemented it. Please review at your convenience, and thanks again for your help with this.
